Prog/vacant-lots-fairfax.Rmd

```diff
@@ -664,28 +627,143 @@ There are `r ` observations that did not have a match in the zoning code. Below,
+                    "INDUST W/OTHER INDUST ZON"
+)
+  
+residential_weird <- c("RES ",
+                       "RD",
+                       "MULTI-FAM W/MIX COM/INDUS",
+                       "R-10",
+                       "RTC-10",
+                       "MIX AG AND CONSRV W/RES/A",
+                       "RP",
+                       "RS-16",
+                       "R-15",
+                       "RS-10",
+                       "MULTI-FAM W/MIX OTHER RES",
+                       "RM-1",
+                       "RS-12.5",
+                       "MULTI FAM W/MIX MULTI FAM",
+                       "R10",
+                       "RES W OTHER RES ZONING",
+                       "RM-2"
+)
+
+
+tempTable<-anti_join(jurfinal, zoningcode, by = c("zoning_code"="code")) %>% 
+  count(zoning_code) %>% arrange(desc(n)) %>% mutate(type = case_when(
+                                            zoning_code %in% planned ~ "planned",
+                                            zoning_code %in% notResidential ~ "not residential",
+                                            zoning_code %in% residential_weird ~ "residential weird",
+                                            T ~ "other"))
+
+tempTable %>% group_by(type) %>% summarise(totalLots = sum(n, na.rm=T))%>%
+  knitr::kable()
+
+```
+
+Of those that do not match, 92% (107,951) are in planning development districts, some of which have residential uses. However, developers must pass the development plan process before building in these districts.  
+```{r results = F}
+
+
+planned_vacants<- jurfinal %>% filter(zoning_code%in%planned, 
+                    category=="vacant") %>%
+                   nrow()
+```
+
+There are `r planned_vacants` vacant lots in the aforementioned planned jurisdictions
+
+
+
 
 
 Filter out lots that are too small to build on and lots with missing zoning information, and select most dense zoning code for those that are big enough.
 
 ```{r filter-small}
 
+
 zoningdata1 <- zoningdata %>% 
-  filter(!is.na(zoning),
-         lotsize_sf >= density,
-         is.na(lot_area)|lotsize_sf >=lot_area,
+  filter(!is.na(newZoning),
+          #lotsize_sf >= density,
+         # is.na(lot_area)|lotsize_sf >=lot_area,
          !grepl("KATELYN ZINN PL", propaddress)) %>% 
-  group_by(propaddress) %>% 
+  mutate(smallsize_flag = ifelse(lotsize_sf >= density, 0, 1)) %>% 
+  mutate(smallsize_flag = ifelse(is.na(smallsize_flag), 0, smallsize_flag)) %>% 
+  group_by(propaddress, smallsize_flag) %>% 
   slice(which.min(dwelling_type)) %>% 
+  ungroup() %>% 
+  group_by(propaddress) %>% 
+  slice(which.min(smallsize_flag)) %>% 
   ungroup() %>%
-  mutate(lotsize_sf = lotsize_sf %>% as.numeric())
+  mutate(lotsize_sf = as.numeric(lotsize_sf),
+         lot_area_min = gsub(lot_area, pattern=",", replacement = "") %>% as.numeric) %>%
+  filter(lotsize_sf>=lot_area_min|is.na(lot_area_min))
+
+# check observations
+nrow(zoningdata1) + nrow(filter(zoningdata, is.na(zoning_code)))
```

Comment:
getting a "NAs introduced by coercion" [1] 200680.  Don't remember seeing this on other counties.

Reply:
Fixed